### PR TITLE
Add an adhoc fairing to include additional headers in responses

### PR DESCRIPTION
**Pull request by fsck**
Addresses #33

### CODE REVIEW

**Review by Trolldemorted — DISMISSED** [review]
Lgtm, but I am certainly not an expert on http security headers, @domenukk should have a say

File: src/main.rs [hunk 1]
```diff
@@ -273,6 +274,11 @@ fn main() {
         .mount("/keyfiles", StaticFiles::from("keyfiles"))
         .mount("/", routes![index_post, index_get, deploy_get, deploy_post, favicon])
         .attach(Template::fairing())
+        .attach(AdHoc::on_response("Security Headers", |_, resp| {
+            resp.adjoin_raw_header("x-xss-protection", "1");
```
Comment (domenukk) [inline src/main.rs]: AFAIK Headers are Case Sensitive.
Also we would want to add the block directive here:
("X-Xss-Protection", "1; mode=block");

File: src/main.rs [hunk 2]
```diff
@@ -273,6 +274,11 @@ fn main() {
...
+            resp.adjoin_raw_header("x-frame-options", "SAMEORIGIN");
```
Comment (domenukk) [inline src/main.rs]: "X-Frame-Options" < case sensitive

File: src/main.rs [hunk 3]
```diff
@@ -273,6 +274,11 @@ fn main() {
...
+            resp.adjoin_raw_header("Content-Security-Policy", "default-src 'self'");
+        }))
```
Comment (domenukk) [inline src/main.rs:289]: Two other handy headers:
("X-Content-Type-Options", "nosniff"); // Disables content sniffing for older browsers
("Referrer-Policy", "no-referrer-when-downgrade"); // won't send a referrer when going from https to http

**Review by domenukk — CHANGES_REQUESTED** [review]
I added some comments about the headers.
AFAIK they are case sensitive... Also we can add more :)

File: src/main.rs (hunk 1 above)
Comment (fsck) [reply src/main.rs]: In your link in #33 they have [example code](https://www.keycdn.com/blog/http-security-headers#2-x-xss-protection) in all lowercase.
[Mozilla](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection) is referring to like this: "X-XSS-Protection: 1"

Comment (domenukk) [reply src/main.rs]: I stand corrected. Still we want mode=block.

File: src/main.rs (hunk 2 above)
Comment (domenukk) [reply src/main.rs]: Apparently not. I stand corrected.

Comment (fsck) [reply src/main.rs]: Each header field consists of a name followed by a colon (":") and the field value. Field names are case-insensitive. [rfc2616](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2)

File: src/main.rs (hunk 3 above)
Comment (fsck) [reply src/main.rs:289]: Sounds good, I will add them to the PR

**Review by domenukk — DISMISSED** [review]

File: src/main.rs [hunk 4]
```diff
@@ -279,6 +280,13 @@ fn main() {
         .mount("/keyfiles", StaticFiles::from("keyfiles"))
         .mount("/", routes![index_post, index_get, deploy_get, deploy_post, favicon])
         .attach(Template::fairing())
+        .attach(AdHoc::on_response("Security Headers", |_, resp| {
+            resp.adjoin_raw_header("X-XSS-Protection", "1");
```
Comment (domenukk) [inline src/main.rs]: Still no "mode=block"

**Review by domenukk — DISMISSED** [review]
looks good

File: src/main.rs (hunk 4 above)
Comment (fsck) [reply src/main.rs]: Should be fixed now

Comment (fsck) [reply src/main.rs]: Sorry for the multiple force-pushes

**Review by Trolldemorted — APPROVED** [review]

**Pull request merged**